### PR TITLE
docs(firestore): emulator documentation

### PR DESCRIPTION
**Pull request by russellwheatley**
Firestore Emulator documentation.

### CODE REVIEW

File: docs/firestore/emulator.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+---
+title: Using The Emulator With Firestore
```
Comment (Salakar) [inline docs/firestore/emulator.md:2]:
```suggestion
title: Using Cloud Firestore Emulator
```

File: docs/firestore/emulator.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+description: Using the emulator to test your app locally
```
Comment (Salakar) [inline docs/firestore/emulator.md:3]:
```suggestion
description: Using the Cloud Firestore emulator to test your app locally.
```

File: docs/firestore/emulator.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+next: /firestore/usage-with-flatlists
+previous: /firestore/usage
+---
+
+You can test your app and its Firestore implementation with an emulator which is built to mimic the behavior of Cloud Firestore. This means you can connect your app directly to the emulator to perform integration testing or QA without touching production data.
+
+For example, you can connect your app to the emulator to safely read and write documents in testing.
+
+# Running the emulator
```
Comment (Salakar) [inline docs/firestore/emulator.md:12]:
```suggestion
## Running the emulator
```

File: docs/firestore/emulator.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+```jsx
+import '@react-native-firebase/app';
+import firestore from '@react-native-firebase/firestore';
+
+const db = firestore();
+
+// set the host property to connect to the emulator
+// set these before any read/write operations occur to ensure it doesn't affect your Cloud Firestore data!
+
+db.settings({ host: 'http://localhost:8080' });
```
Comment (Salakar) [inline docs/firestore/emulator.md:39]: Is with the `http` protocol part correct? Though it was without on the previous docs you had?

If it's without then this and js replacer code below would need changing.

Comment (russellwheatley) [reply docs/firestore/emulator.md:39]: I tried to make it consistent with how we approached it in cloud functions emulator which passes along the `http`protocol. AlthoughI have changed what we pass under the hood to Android. If you pass along the `http` protocol, the app crashes.

**Comment by Ehesp** [comment]
Just thinking, does this need to have `http://`? it should automatically handle the protocol with the ssl flag.

**Comment by russellwheatley** [comment]
@ehesp, I don't think  either iOS or Android need the http protocol prepended: https://firebase.google.com/docs/emulator-suite/connect_and_prototype#instrument_your_app_to_talk_to_the_emulators. Would you like me to remove it's necessity from JS side (& docs)?

**Comment by Salakar** [comment]
> @Ehesp, I don't think either iOS or Android need the http protocol prepended: [firebase.google.com/docs/emulator-suite/connect_and_prototype#instrument_your_app_to_talk_to_the_emulators](https://firebase.google.com/docs/emulator-suite/connect_and_prototype#instrument_your_app_to_talk_to_the_emulators). Would you like me to remove it's necessity from JS side (& docs)?

Lets remove it, worth confirming everything works afterwards too if possible

**Comment by russellwheatley** [comment]
Removed `http` protocol. Tested on `iOS` & `Android`, @Salakar

**Pull request merged**